src/viewProviders/resources.ts

```diff
@@ -201,6 +204,12 @@ export class ResourceViewProvider implements vscode.TreeDataProvider<ResourceVie
             env.isLoading = cachedLoading;
             this.cachedLoadingStates.delete(env.id);
           }
+          const latestStatus: ConnectionStatus | undefined = watcher.getLatestConnectionEvent(
```

Comment:
Probably need to comment intent / how we ended up here? So many twists and turns.

Reply:
Ah true, we commented the part inside `refreshConnection()` but not this one. Added in https://github.com/confluentinc/vscode/pull/946/commits/e02f05b78c72a4c1f75aff1e77c301136e72e442